code/modules/mob/mob.dm

```diff
@@ -2060,7 +2060,7 @@ mob/proc/on_foot()
 	spawn(duration + 1)
 		regenerate_icons()
 
-/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE)	//transforms the mob into a new member of the given mob type, while preserving the mob's body
+/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE, var/kill_on_death = TRUE, var/allow_revert = TRUE)	//transforms the mob into a new member of the given mob type, while preserving the mob's body
```

Comment:
Why add an argument for making a transmogrification permanent? All of the utility is in being able to undo the transformations via code if necessary.
All of the code that is run when `allow_revert` is `TRUE` is essentially
```
var/mob/M = new type(loc)
M.key = key
qdel(src)
```
In which case you could just use those lines instead, or make a new proc specifically to permanently move a player into a new instance of a given mob if you'd really rather use 1 line than 3.